app/src/androidTestKiwix/java/org/kiwix/kiwixmobile/tests/SettingsTest.java

```diff
@@ -0,0 +1,135 @@
+package org.kiwix.kiwixmobile.tests;
+
+import android.preference.Preference;
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kiwix.kiwixmobile.R;
+import org.kiwix.kiwixmobile.utils.SplashActivity;
+
+import static android.support.test.espresso.Espresso.onData;
+import static android.support.test.espresso.Espresso.onView;
+import static android.support.test.espresso.action.ViewActions.click;
+import static android.support.test.espresso.assertion.ViewAssertions.matches;
+import static android.support.test.espresso.matcher.PreferenceMatchers.withKey;
+import static android.support.test.espresso.matcher.ViewMatchers.isDisplayed;
+import static android.support.test.espresso.matcher.ViewMatchers.withText;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.instanceOf;
+import static org.hamcrest.Matchers.is;
+import static org.kiwix.kiwixmobile.utils.StandardActions.enterSettings;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+public class SettingsTest {
+  @Rule
+  public ActivityTestRule<SplashActivity> mActivityTestRule = new ActivityTestRule<>(
+      SplashActivity.class);
```

Comment:
Why is the rule for ```SplashActivity``` instead of ```KiwixSettingsActivity``` ?

Reply:
@srv-twry The "ContentTest" does the same, I thought that is the expected way.

Reply:
Oh! okay. I am not that experience with Espresso, so may be that's the correct way.